cadquery/cq.py

```diff
@@ -1916,6 +1916,41 @@ def makeCircleWire(obj):
 
         return self.eachpoint(makeCircleWire, useLocalCoordinates=True)
 
+    # ellipse from current point
+    def ellipse(self, x_radius, y_radius, angle1=360, angle2=360, rotation_angle=0.0,
```

Comment:
Wouldn't angle1=0  be more logical?

Reply:
absolutely, however, makeCircle (and hence makeEllipse) uses 360 and I decided to keep that up to the top level. And from the later logic perspective, angle1 needs to be equal angle2 to get a full circle.

Reply:
... or full ellipse

Reply:
I see that circle does not expose those options at all - @jmwright what do you think?

Reply:
I just looked at `Workplane.spline` which follows a different approach than `Workplane.circle`:
It does only have `Edge.makeSpline` and then applies the following logic
https://github.com/CadQuery/cadquery/blob/74573fc3bb032321c50a9e817dbf806c2a286d53/cadquery/cq.py#L1441-L1450

This allows splines to be used when building 2D paths.

If we expose `angle1` and `angle2` with `ellipse` shouldn't we change it like `spline` to allow ellipse arcs being used in building 2D paths? That was my original intention with exposing both angles, but I missed the point that `circle` doesn't add itself to pendingWires/pendingEdges (and hence my `ellipse` code doesn't either.

Reply:
I wouldn't assume that the code in `circle` is the correct way to do it. I don't think that code has been touched since the early days of CQ, and may not represent our current understanding of best practices.